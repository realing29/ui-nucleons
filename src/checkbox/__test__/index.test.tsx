import React from 'react';
import { mount } from 'enzyme';
import { Checkbox } from '..';

describe('<Checkbox />', () => {
  it('should render without props', () => {
    const wrapper = mount(<Checkbox />);

    expect(wrapper.find('input')).toHaveLength(1);
    expect(wrapper.find('input').prop('type')).toBe('checkbox');
  });

  it('should renders as disabled', () => {
    const wrapper = mount(<Checkbox disabled />);

    expect(wrapper.find('input')).toHaveLength(1);
    expect(wrapper.find('input').prop('disabled')).toBe(true);
  });

  it('should handle "defaultChecked" prop', () => {
    const wrapper = mount(<Checkbox defaultChecked />);

    expect(wrapper.find('input').prop('checked')).toBe(true);
  });

  it('should handle "id" property', () => {
    const wrapper = mount(<Checkbox id='test-checkbox' />);

    expect(wrapper.find('input').prop('id')).toBe('test-checkbox');
  });

  it('should handle "onChange" property', () => {
    const spy = jest.fn();

    const wrapper = mount(<Checkbox onChange={spy} />);

    expect(spy).toHaveBeenCalledTimes(0);

    wrapper.find('input').simulate('change');
    expect(spy).toHaveBeenCalledTimes(1);

    wrapper.setProps({ onChange: null });
    expect(() => wrapper.find('input').simulate('change')).not.toThrow();
  });

  it('should renders correctly with props', () => {
    const spy = jest.fn();
    const wrapper = mount(<Checkbox checked onChange={spy} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.checkbox-container.error')).toHaveLength(0);
  });

  it('should pass "className" prop', () => {
    const wrapper = mount(<Checkbox className='test-class-name' />);

    expect(wrapper.find('.root').prop('className')).toContain('test-class-name');
  });
});
