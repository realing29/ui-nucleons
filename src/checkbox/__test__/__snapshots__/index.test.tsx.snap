// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should renders correctly with props 1`] = `
<Checkbox
  checked={true}
  onChange={[MockFunction]}
>
  <span
    className="checkbox-wrapper"
  >
    <input
      checked={true}
      className="checkbox-input"
      onChange={[Function]}
      type="checkbox"
    />
    <SvgCheckbox
      className="checkbox-icon checked"
    >
      <svg
        className="checkbox-icon checked"
        height={16}
        viewBox="0 0 16 16"
        width={16}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <rect
            fill="#fff"
            height={15}
            rx={1.5}
            stroke="#9E9E9E"
            width={15}
            x={0.5}
            y={0.5}
          />
        </g>
        <g>
          <rect
            fill="#1F84DB"
            height={16}
            rx={2}
            width={16}
          />
          <path
            d="M10.8 5.22L6.4 9.19l-1.034-.97a.838.838 0 00-1.132 0 .717.717 0 000 1.06l1.6 1.5a.838.838 0 001.132 0l4.8-4.5a.717.717 0 000-1.06c-.313-.293-.654-.293-.966 0z"
            fill="#fff"
          />
        </g>
        <g>
          <rect
            fill="#F5F5F5"
            height={15}
            rx={1.5}
            stroke="#C2C2C2"
            width={15}
            x={0.5}
            y={0.5}
          />
        </g>
        <g
          opacity={0.48}
        >
          <rect
            fill="#757575"
            height={16}
            rx={2}
            width={16}
          />
          <path
            d="M10.8 5.22L6.4 9.19l-1.034-.97a.838.838 0 00-1.132 0 .717.717 0 000 1.06l1.6 1.5a.838.838 0 001.132 0l4.8-4.5a.717.717 0 000-1.06c-.313-.293-.654-.293-.966 0z"
            fill="#fff"
          />
        </g>
      </svg>
    </SvgCheckbox>
  </span>
</Checkbox>
`;
